Data/Aeson/Types/FromJSON.hs

```diff
@@ -1260,15 +1260,15 @@ instance (ProductFromJSON arity f, ProductSize f
 --------------------------------------------------------------------------------
 
 class FieldNames f where
-    fieldNames :: f a -> [Text] -> [Text]
+    fieldNames :: f a -> [String] -> [String]
```

Comment:
I changed the type signature from `[Text]` to `[String]` deliberately to be able to call the `fieldLabelModifier` without a `pack` / `unpack` cycle.